prov/usnic/src/usdf_fabric.c

```diff
@@ -117,7 +117,8 @@ usdf_validate_hints(struct fi_info *hints, struct usd_device_attrs *dap)
 			return -FI_ENODATA;
 		}
 		if (fattrp->name != NULL &&
-                    strcmp(fattrp->name, dap->uda_devname) != 0) {
+                    strcmp(fattrp->name, dap->uda_devname) != 0) { // CHANGE ??
```

Comment:
I think you need to change this `strcmp()` to be based on the version, just like the others.  For <=v1.3, the current `strcmp()` is fine.  For >=v1.4, make up a CIDR string and compare against that.

Double check the sockets provider, but I assume they are doing something similar (at least for >=v1.4).